### ISSUE TITLE
[Prep] EX: Remove the account migration screen

### ISSUE THREAD

**Issue by manuroe** [vector-im/element-meta#2497]
### Description

With SSS, there will be no more a SS proxy in the middle that requires to do a super slow initial v2 `/sync` for the very first login. We can remove this screen from the app.

[Internal discussion](https://matrix.to/#/!cnVVNLKqgUzNTOFQkz:matrix.org/$TMMp2eSrY8zJcQUBtg6fUmJLfR6X29u0UwXoys-iHiM?via=element.io&via=matrix.org&via=jki.re)

<details>
<summary>The screen to kill</summary>
<img src="https://github.com/user-attachments/assets/ed006eb7-48ed-4fcd-9a78-4a4b18732b70"/>
</details>

# Acceptance criteria
- The screen does not exist anymore in the code
- The app directly displays the room list

# Size estimate
S/M/L

# Dependencies    
**We must release this in EX apps only after the synapse release and global SSS availability on matrix.org**.

# Out of scope
- 

# Subtasks
```[tasklist]
### Android
- [ ] https://github.com/element-hq/element-x-android/issues/3292
```

```[tasklist]
### iOS
- [ ] https://github.com/element-hq/element-x-ios/issues/3146
```
